New release v3.0.0

holgerd77 opened the pull request:
Already prepared the release notes for ``v3.0.0``, feel free to comment everyone.

holgerd77 commented:
@s1na Can you do a review here now?

s1na commented on CHANGELOG.md:
```diff
@@ -6,6 +6,89 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+
+This release comes with a modernized ``ES6``-class structured code base as well
+as some significant local refactoring work regarding how ``Stack`` and ``Memory``
+are organized within the VM execution engine. It also finalizes the module-structuring
```
> It also finalizes the module-structuring

Hopefully this doesn't mean the door for future module-structuring is closed after this release? :smile:

holgerd77 replied:
Have toned down a bit. 😛

s1na commented on CHANGELOG.md:
```diff
@@ -6,6 +6,89 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+  
+**Code Modernization/ES6**
+
+- Converted ``VM`` to ``ES6`` class,
```
This PR also drops the `VM.deps` attribute. I don't think anyone was using that, but it might be worth a mention.

holgerd77 replied:
Have added a note (generally: will push after my comments)

s1na commented on CHANGELOG.md:
```diff
@@ -6,6 +6,89 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+
+- Fixed a bug where ``stateManager.setStateRoot()`` didn't clear 
+  the ``_storageTries`` cache,
+  PR [#444](https://github.com/ethereumjs/ethereumjs-vm/issues/444)
```
Nitpick: This links to the issue, the PR is #445 :)

holgerd77 replied:
Updated.

s1na commented on CHANGELOG.md:
```diff
@@ -6,6 +6,89 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+  PR [#462](https://github.com/ethereumjs/ethereumjs-vm/pull/462)
+- Fixed ``ecmul_0-3_5616_28000_96`` (by test setup adoption),
+  PR [#473](https://github.com/ethereumjs/ethereumjs-vm/pull/473)
+
```
Another PR thanks to @chikeichan: #438

holgerd77 replied:
I also referenced the issue here and named it "PR" 😄, corrected.

s1na commented:
Another comment related to #484: the default HF in `VM` is still `Byzantium`. Should we maybe change that to `Petersburg` for this release?

holgerd77 commented:
That's a super-good catch with ``Byzantium`` -> ``Petersburg``, makes very much sense and would have very much been a pity if missed the chance on this release, since timing is perfect! 😄

Can you prepare a PR for that?

holgerd77 commented:
Ok, updated, this is now ready for a re-review. Have also cleaned up and updated the README - especially according to the HF default update - but also on other things which were outdated/becoming incorrect.
